jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/ServletContextHandlerTest.java

```diff
@@ -1852,14 +1853,23 @@ public void contextDestroyed(ServletContextEvent sce)
                     contextDestroy.set(true);
             }
         });
-        ServletHandler shandler = servletContextHandler.getServletHandler();
 
-        ResourceHandler rh = new ResourceHandler();
+        org.eclipse.jetty.server.handler.ResourceHandler coreResourceHandler = new org.eclipse.jetty.server.handler.ResourceHandler();
+        HandlerWrapper nestedHandler = new HandlerWrapper();
+        Handler last = new HandlerWrapper();
+
+        servletContextHandler.insertHandler(coreResourceHandler);
+        servletContextHandler.insertHandler(nestedHandler);
+        servletContextHandler.getTail().setHandler(last);
```

Comment:
Calling `servletContextHandler.setHandler(last)` seems more natural.

Reply:
@sbordet  With what semantics?   Is that replacing any optional SessionHandler or SecurityHandler that may be included?  Does that replace the ServletHandler that should always be there?   or is it a set on the ServletHandler?  If it is called twice is it an insert on the ServletHandler?

We have always had that a handler.setHandler(next), sets the next of that handler and not some tail or chain etc.     In a few cases we disable setHandler when it is not appropriate.    I'm not sure I like defining new semantics

Reply:
done.